client/web/web_test.go

```diff
@@ -76,14 +76,16 @@ func TestQnapAuthnURL(t *testing.T) {
 //  2. localapi proxy allowlist
 func TestServeAPI(t *testing.T) {
 	lal := memnet.Listen("local-tailscaled.sock:80")
-	defer lal.Close()
 
 	// Serve dummy localapi. Just returns "success".
 	go func() {
 		localapi := &http.Server{Handler: http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 			fmt.Fprintf(w, "success")
 		})}
+
 		defer localapi.Close()
+		defer lal.Close()
```

Comment:
this will never exit.

Reply:
previously the `lal.Close()` was causing the `localapi.Serve` call to return, but it was logging an error. That error log was racing with the test returning. See #9151